metpy/calc/tests/test_tools.py

```diff
@@ -187,3 +187,66 @@ def test_log_interp_units():
     y_interp_truth = np.array([20.0343863828, 24.6395565688, 29.2447267548]) * units.degC
     y_interp = log_interp(x_interp, x_log, y_log)
     assert_array_almost_equal(y_interp, y_interp_truth, 7)
+
+
+def test_get_layer_ragged_data():
+    """Tests that error is raised for unequal length pressure and data arrays."""
+    p = np.arange(10) * units.hPa
+    y = np.arange(9) * units.degC
+    with pytest.raises(ValueError):
+        get_layer(p, y)
+
+
+def test_get_layer_bottom_out_of_range():
+    """Tests that error is raised when the bottom of the layer is out of the data range."""
+    p = np.arange(900, 500, -100) * units.hPa
+    y = np.arange(4) * units.degC
+    with pytest.raises(ValueError):
+        get_layer(p, y, bottom=1000 * units.hPa)
+
+
+def test_get_layer_top_out_of_range():
+    """Tests that error is raised when the top of the layer is out of the data range."""
+    p = np.arange(900, 500, -100) * units.hPa
+    y = np.arange(4) * units.degC
+    with pytest.raises(ValueError):
+        get_layer(p, y, depth=700 * units.hPa)
+
+
+def test_get_layer_inverted_pressures():
+    """Tests that error is raised if pressure at the top of layer is larger than the bottom."""
+    p = np.arange(900, 500, -100) * units.hPa
+    y = np.arange(4) * units.degC
+    with pytest.raises(ValueError):
+        get_layer(p, y, depth=-100 * units.hPa, bottom=600 * units.hPa)
+
+
+p = np.arange(1000, 10, -100) * units.hPa
```

Comment:
I mean that the practice isn't anywhere else. I think I've seen global data elsewhere, but haven't gone looking.